packages/collector/Collector.php

```diff
@@ -0,0 +1,229 @@
+
+function collector_plugins_loaded()
+{
+    if(!current_user_can('manage_options'))
+    {
+        return;
+    }
+
+    if(urldecode($_SERVER['REQUEST_URI']) === COLLECTOR_DOWNLOAD_PATH)
+    {
+        collector_zip_collect();
+        collector_zip_send();
+        collector_zip_delete();
+        exit();
+    }
+}
+
+function collector_plugin_menu()
+{
+    add_submenu_page(
+        NULL,
+        'Collector',
+        'Collector',
+        'manage_options',
+        'collector_render_playground_page',
+        'collector_render_playground_page',
+        NULL
+    );
+}
+
+function collector_render_playground_page()
+{?>
+	<div id = "wp-playground-wrapper">
+		<div id = "wp-playground-toolbar">
+			NOW WORKING INSIDE WORDPRESS PLAYGROUND &nbsp; [<a href = "/wp-admin" id = "goBack">RETURN</a>]
+		</div>
+		<div id = "wp-playground-main-area">
+			<iframe id = "wp-playground"></iframe>
+		</div>
+	</div>
+	<script type = "text/javascript">
+		const frame  = document.getElementById('wp-playground');
+		const zipUrl = <?=json_encode(COLLECTOR_DOWNLOAD_PATH);?>;
+
+		const username   = <?=json_encode(wp_get_current_user()->user_login);?>;
+		const fakepass   = <?=json_encode(collector_get_fakepass());?>;
+		const pluginUrl  = new URLSearchParams(window.location.search).get('pluginUrl');
+        const pluginName = new URLSearchParams(window.location.search).get('pluginName');
+
+		(async () => {
+			const  { startPlaygroundWeb } = await import(<?=json_encode(COLLECTOR_PLAYGROUND_PACKAGE);?>);
+
+			const steps = [
+				{
+					step: 'writeFile',
+					path: '/data.zip',
+					data: {
+						'resource': 'url',
+						'url': zipUrl,
+					},
+				},
+				{
+					step: 'unzip',
+					zipPath: '/data.zip',
+					extractToPath: '/wordpress',
+				},
+				{
+					step: 'rm',
+					path: '/data.zip',
+				},
+				{
+					step: 'runSqlFile',
+					path: '/wordpress/schema/_Schema.sql',
+				},
+				{
+					step: 'rm',
+					path: '/wordpress/wp-content/mu-plugins/1-show-admin-credentials-on-wp-login.php',
+				},
+			];
+
+			if(pluginUrl && pluginName)
+			{
+				steps.push(...[
+					{
+						step: 'writeFile',
+						path: '/plugin.zip',
+						data: {
+							'resource': 'url',
+							'url': pluginUrl,
+						},
+					},
+					{
+						step: 'unzip',
+						zipPath: '/plugin.zip',
+						extractToPath: '/wordpress/wp-content/plugins',
+					},
+					{
+						step: 'rm',
+						path: '/plugin.zip',
+					},
+					{
+						step: 'activatePlugin',
+						pluginName: pluginName,
+						pluginPath: '/wordpress/wp-content/plugins/' + pluginName,
+					},
+				]);
+			}
+
+			steps.push(...[
+				{
+					step: 'login',
+					username: username,
+					password: fakepass,
+				}
+			]);
+
+			const client = await startPlaygroundWeb({
+				iframe: document.getElementById('wp-playground'),
+				remoteUrl: `https://playground.wordpress.net/remote.html`,
+				blueprint: {
+					preferredVersions: {
+						wp: <?=json_encode(COLLECTOR_WP_VERSION);?>,
```

Comment:
I just checked and this actually works with `short_open_tag=0`. PHP is weird!

Reply:
It's because `short_open_tag` in PHP is `<?`.

`<?=` is not a short open tag, it's something else used mostly in templating.

Reply:
I was so happy on March 1st, 2012: https://www.php.net/ChangeLog-5.php#5.4.0